### ISSUE TITLE
progress for individually-assigned learners not showing on coach reports

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#6341]
### Observed behavior

observed by @jonboiser 

> Progress for individually-assigned learner not showing on coach reports
> 
> e.g. in [http://kolibribeta.learningequality.org/en/coach/#/612355f5712f26fb842aecf17fb2c376/reports/lessons/f10ae0ca4dc9f5a73bbe1469b1db3ffd/resources](http://kolibribeta.learningequality.org/en/coach/#/612355f5712f26fb842aecf17fb2c376/reports/lessons/f10ae0ca4dc9f5a73bbe1469b1db3ffd/resources)
> Jamal Khan isn't being counted in the progress reports.

### Expected behavior

all progress shows

### User-facing consequences

coaches miss critical information

### Steps to reproduce

see link above

### Context

0.13.0 beta 6

**Comment by nucleogenesis** [comment]
@indirectlylit @rtibbles 

An error is occurring when progress is being saved.

For example, when I have a certain FacilityUser `user` and I call `user.memberships.all()`

I am getting:

```
Traceback (most recent call last):
  File "/home/jacob/.local/share/virtualenvs/kolibri--v441dVg/lib/python3.5/site-packages/django/db/models/fields/related_descriptors.py", line 178, in __get__
    rel_obj = getattr(instance, self.cache_name)
AttributeError: 'Membership' object has no attribute '_collection_cache'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/jacob/.local/share/virtualenvs/kolibri--v441dVg/lib/python3.5/site-packages/django/db/models/query.py", line 229, in __repr__
    return '<%s %r>' % (self.__class__.__name__, data)
  File "/home/jacob/.local/share/virtualenvs/kolibri--v441dVg/lib/python3.5/site-packages/django/db/models/base.py", line 590, in __repr__
    u = six.text_type(self)
  File "/home/jacob/Code/LearningEquality/kolibri/kolibri/core/auth/models.py", line 1173, in __str__
    user=self.user, collection=self.collection
  File "/home/jacob/.local/share/virtualenvs/kolibri--v441dVg/lib/python3.5/site-packages/django/db/models/fields/related_descriptors.py", line 184, in __get__
    rel_obj = self.get_object(instance)
  File "/home/jacob/.local/share/virtualenvs/kolibri--v441dVg/lib/python3.5/site-packages/django/db/models/fields/related_descriptors.py", line 159, in get_object
    return qs.get(self.field.get_reverse_related_filter(instance))
  File "/home/jacob/.local/share/virtualenvs/kolibri--v441dVg/lib/python3.5/site-packages/django/db/models/query.py", line 380, in get
    self.model._meta.object_name
kolibri.core.auth.models.DoesNotExist: Collection matching query does not exist.
```

I got that because I found that it is choking here:
https://github.com/learningequality/kolibri/blob/ae47e83cc8f9c2a31d95f5915bb2e8edaed3cd3e/kolibri/core/notifications/api.py#L25 

The strange thing is that this doesn't happen for every user. The user for whom it does happen in my testing has Memberships in the DB as well. 

I can't find anything in the code (or elsewhere) mentioning `_collection_cache` either.

**Comment by rtibbles** [comment]
I am fairly sure it's an internal field used by Django to cache related table lookups.

Does it consistently happen for the user in question?

**Comment by ralphiee22** [comment]
After some digging we found that the `Collection` the `Membership` model is trying to reference does not actually exist, hence the errors above. When deleting a `Collection` model, it should cascade delete the `Membership` model. Either the DB somehow got in a bad state or there is a bug in the code. Being able to reproduce the error above most likely means it's the latter.

**Comment by nucleogenesis** [comment]
@rtibbles @ralphiee22 

The issue I was running into was due to my own local DB being screwed up. Likely related to my own poking around.

I've not been able to replicate _that specific_ error on a fresh DB after a lot of trying to break things with normal use. Basically - I had some extra memberships lying around in the DB pointing at a nonexistent collection. 

I'm pretty sure I deleted some `adhoclearnersgroup` collections in the sqlite browser during testing that caused this locally.

The actual issue here is that I didn't account for how the notifications are collated in the API and then summarized on the frontend etc. Fixed in imminent PR.
